websites/D/Discord Turkiye/presence.ts

```diff
@@ -0,0 +1,120 @@
+const presence = new Presence({ clientId: "890757020393816064" }),
+	startTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "1024dctr",
+			details: "Geziniyor",
+			startTimestamp,
+		},
+		{ pathname } = document.location,
+		privacyMode = await presence.getSetting("privacy_mode").catch(() => false);
+
+	if (document.querySelector('[href="/login/"]')) {
+		presenceData.smallImageText = "Misafir Modu";
+		presenceData.smallImageKey = "https://i.hizliresim.com/385dt7x.png";
+	} else if (document.querySelector("span.avatar.avatar--xxs > img")) {
+		presenceData.smallImageText = document
+			.querySelector("span.avatar.avatar--xxs > img")
+			.getAttribute("alt");
+		presenceData.smallImageKey = "https://i.hizliresim.com/385dt7x.png";
+	}
+
+	switch (pathname) {
+		case "/":
+			presenceData.details = "Anasayfayı görüntülüyor";
+			break;
+		case "/hesabim":
+			presenceData.details = "Kişisel detaylarını görüntülüyor";
+			break;
+		case "/kullanicilar/":
+			presenceData.details = "Kullanıcıları görüntülüyor";
+			break;
+		case "/araclar/":
+			presenceData.details = "Araçları görüntülüyor";
+			delete presenceData.smallImageKey;
+			break;
+
+		case "/araclar/davet/":
+			delete presenceData.smallImageKey;
+			if (privacyMode) presenceData.details = "Davet görüntülüyor";
+			else if (
+				document.querySelector<HTMLElement>("div.tag > div.name").textContent
+			) {
+				presenceData.details = "Daveti görüntülüyor:";
+				presenceData.state =
+					document.querySelector<HTMLElement>("div.tag > div.name").textContent;
+				presenceData.largeImageKey = document
+					.querySelector<HTMLElement>("div.icon")
+					.style.backgroundImage.split('url("')[1]
+					.split('");')[0];
+			}
+			break;
+		case "/konular/":
+			presenceData.details = "Konuyu görüntülüyor:";
+			presenceData.state = document.title.split(" | ")[0];
+			if (
+				document.querySelector<HTMLElement>("div.fr-element.fr-view > p")
+					?.textContent !== "\n"
```

Comment:
You had a truthy check before, this is safer
```suggestion
					?.textContent.trim()
```